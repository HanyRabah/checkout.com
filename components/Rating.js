import styled from 'styled-components'
import Stars from './Stars'
import Tooltip from './Tooltip'

const Rating = (props) => {
  const {
    label,
    value,
    handleChange,
    errorText,
    stars,
    readOnly,
    required,
  } = props
  const totalStars = stars ? stars : 5

  return (
    <Container>
      <Label>
        {required && <Required title="required">*</Required>} {label}
      </Label>
      <RatingContainer>
        {Array(totalStars)
          .fill(null)
          .map((_, i) => i + 1)
          .map((starValue) => (
            <Star key={`rating-${starValue}`}>
              <Radio
                type="radio"
                name="rating"
                value={starValue}
                disabled={readOnly}
                onChange={handleChange}
              />
              <Stars selected={starValue <= value} />
            </Star>
          ))}
      </RatingContainer>
      {errorText && <Tooltip text={errorText} />}
    </Container>
  )
}

const Container = styled.div`
  position: relative;
`

const Label = styled.p`
  font-size: 14px;
  margin-bottom: 8px;
  font-weight: 600;
  position: relative;
`

const UList = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
  display: flex;
`
const RatingContainer = styled.div`
  display: flex;
`

const Star = styled.div`
  position: relative;
`

const Radio = styled.input`
  cursor: pointer;
  position: relative;
  opacity: 0;
  width: 36px;
  height: 36px;
  position: absolute;
  cursor: ${(props) => (props.disabled ? 'auto' : 'pointer')};
`

const ValidationMessage = styled.span`
  color: red;
  font-size: 11px;
`
const Required = styled.span`
  color: #fd496d;
`

export default Rating
