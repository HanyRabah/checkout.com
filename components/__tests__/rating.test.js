import React from 'react';
import Rating from '../Rating';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';

describe('Rating', () => {
    afterEach(cleanup);
  
    it('render correctly', () => {
      const component = renderer.create(<Rating />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
});