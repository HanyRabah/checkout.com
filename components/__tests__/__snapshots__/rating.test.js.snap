// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating render correctly 1`] = `
<div
  className="Rating__Container-sc-8fqffi-0 BzQdw"
>
  <p
    className="Rating__Label-sc-8fqffi-1 iCfwlL"
  >
     
  </p>
  <div
    className="Rating__RatingContainer-sc-8fqffi-3 eMkkQo"
  >
    <div
      className="Rating__Star-sc-8fqffi-4 gIripz"
    >
      <input
        className="Rating__Radio-sc-8fqffi-5 cfQPPy"
        name="rating"
        type="radio"
        value={1}
      />
      <svg
        className="Stars__Star-agauv2-0 htjMpb"
        selected={false}
        viewBox="0 0 36 34"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="Rating__Star-sc-8fqffi-4 gIripz"
    >
      <input
        className="Rating__Radio-sc-8fqffi-5 cfQPPy"
        name="rating"
        type="radio"
        value={2}
      />
      <svg
        className="Stars__Star-agauv2-0 htjMpb"
        selected={false}
        viewBox="0 0 36 34"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="Rating__Star-sc-8fqffi-4 gIripz"
    >
      <input
        className="Rating__Radio-sc-8fqffi-5 cfQPPy"
        name="rating"
        type="radio"
        value={3}
      />
      <svg
        className="Stars__Star-agauv2-0 htjMpb"
        selected={false}
        viewBox="0 0 36 34"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="Rating__Star-sc-8fqffi-4 gIripz"
    >
      <input
        className="Rating__Radio-sc-8fqffi-5 cfQPPy"
        name="rating"
        type="radio"
        value={4}
      />
      <svg
        className="Stars__Star-agauv2-0 htjMpb"
        selected={false}
        viewBox="0 0 36 34"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="Rating__Star-sc-8fqffi-4 gIripz"
    >
      <input
        className="Rating__Radio-sc-8fqffi-5 cfQPPy"
        name="rating"
        type="radio"
        value={5}
      />
      <svg
        className="Stars__Star-agauv2-0 htjMpb"
        selected={false}
        viewBox="0 0 36 34"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
